docs/apis/subsystems/tag/index.md

```diff
@@ -0,0 +1,274 @@
+
+There are more options such as specifying the default value for "Standard tags", having a fixed collection or excluding from search. They can be found in comments in [//github.com/moodle/moodle/blob/master/lib/db/tag.php lib/db/tag.php](https://docs.moodle.org/https///github.com/moodle/moodle/blob/master/lib/db/tag.php_lib/db/tag.php)
+
+After making changes to db/tag.php plugin developer must bump the plugin version in **version.php** and run upgrade script. This usually applies to any changes in the db/ folder.
+
+### Adding tags element to the editing form
+
+After the tag area is defined it should appear on the "Manage tags" page. Now it is time to allow users to add/change tags when editing the item. Here is an example from Wiki module:
+
+1. Add a 'tags' form element to the editing form:
+
+ ```php
+ $mform->addElement('tags', 'tags', get_string('tags'), array('itemtype' => 'wiki_pages', 'component' => 'mod_wiki'));
+ ```
+
+ This element will automatically check if the tag area is disabled by the manager and will not display anything in this case. However if you want to add a header you need to check if tag area is enabled add this:
+
+ ```php
+ if (core_tag_tag::is_enabled('mod_wiki', 'wiki_pages')) {
+     $mform->addElement('header', 'tagshdr', get_string('tags', 'tag'));
+ }
+ ```
+
+Foobar
+2. Save the form data
+
+ ```php
+ if ($data = $form->get_data()) {
+    // Do some other processing here, if this is a new page (item) you need to insert it in the DB and obtain id.
+    // $pageid = $data->id;
+    core_tag_tag::set_item_tags('mod_wiki', 'wiki_pages', $pageid, $modulecontext, $data->tags);
+ }
+ ```
+
+ It is important to specify the correct context in this function. Note that $data->tags will always be returned by the form, even if the area is disabled, however core_tag_tag::set_item_tags() will not change or reset tags if the tag area is disabled.
+3. Populate the form with existing tags before calling $form->set_data($data):
+
+ ```php
+ $data = $DB->get_record('wiki_pages', array('id' => $this->page->id)); // Well, it's more complicated than that of course....
+ $data->tags = core_tag_tag::get_item_tags_array('mod_wiki', 'wiki_pages', $this->page->id);
+ $form->set_data($data);
+ ```
+
+ Always test the code with tag area enabled and disabled.
+
+### Displaying tags next to the item
+
+Example of displaying of the tags are user interests on the user profile page. User can see the list of interests, each of them is a link that leads to the page that shows all items tagged with this tag.
+
+Here is the code used by Wiki module to display tags the page is tagged with:
+
+```php
+echo $OUTPUT->tag_list(core_tag_tag::get_item_tags('mod_wiki', 'wiki_pages', $page->id), null, 'wiki-tags');
+```
+
+If you will need to load the tags for a lot of different item ids, when use core_tag_tag::get_items_tags to load them all in a single DB query.
```

Comment:
This sentence needs some attention.
At the very least it needs a bit of grammar, and the function call should be in backticks.
I'd also suggest putting it into an admition, (maybe important?)
```suggestion
If you will need to load the tags for a lot of different item ids, when use `core_tag_tag::get_items_tags()` to load them all in a single DB query.
```